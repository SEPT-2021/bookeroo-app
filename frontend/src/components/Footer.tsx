import React from "react";
import { createStyles, withStyles, WithStyles } from "@material-ui/core";

function Footer({ classes }: FooterProps) {
  return (
    <div className={classes.footer}>
      <footer>
        <p> 2021 Â© SEPT Bookeroo. All rights reserved.</p>
      </footer>
    </div>
  );
}

const styles = () =>
  createStyles({
    footer: {
      padding: "10px 20px",
      background: "#666",
      color: "white",
      bottom: "0",
      width: "100%",
      textAlign: "center",
    },
  });

type FooterProps = WithStyles<typeof styles>;

export default withStyles(styles)(Footer);
