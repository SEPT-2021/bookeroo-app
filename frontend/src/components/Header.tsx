import React, { useEffect, useState } from "react";
import {
  Button,
  Collapse,
  createStyles,
  IconButton,
  withStyles,
  WithStyles,
} from "@material-ui/core";
import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
import { Link as Scroll } from "react-scroll";
import TextLoop from "react-text-loop";
import Link from "../util/Link";

function Header({ classes }: HeaderProps) {
  const [checked, setChecked] = useState(false);
  useEffect(() => {
    setChecked(true);
  }, []);
  return (
    <div className={classes.root} id="header">
      <Collapse
        in={checked}
        {...(checked ? { timeout: 1000 } : {})}
        collapsedHeight={50}
      >
        <div className={classes.container}>
          <h1 className={classes.title}>
            Welcome to <br />
            <span className={classes.colorText}>Bookeroo</span>
            <TextLoop
              interval={1000}
              fade
              springConfig={{ stiffness: 200, damping: 5 }}
            >
              <span> . </span>
              <span> ! </span>
            </TextLoop>
          </h1>
          <Link to="/allBooks">
            <Button variant="contained" color="primary">
              View our Collection
            </Button>
          </Link>
          <br />
          <Scroll to="contact" smooth>
            <IconButton>
              <ExpandMoreIcon className={classes.goDown} />
            </IconButton>
          </Scroll>
        </div>
      </Collapse>
    </div>
  );
}

const styles = () =>
  createStyles({
    root: {
      display: "flex",
      justifyContent: "center",
      alignItems: "center",
      height: "100vh",
      fontFamily: "Arial Rounded MT Bold",
    },
    colorText: {
      color: "#5DA2D5",
    },
    icon: {
      color: "#F78888",
      fontSize: "2rem",
    },

    container: {
      textAlign: "center",
    },
    title: {
      color: "#F8E9A1",
      fontSize: "4.5rem",
    },
    goDown: {
      color: "#A8D0E6",
      fontSize: "4rem",
    },
  });

type HeaderProps = WithStyles<typeof styles>;

export default withStyles(styles)(Header);
