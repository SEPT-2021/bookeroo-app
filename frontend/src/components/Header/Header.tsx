import React, {useEffect, useState} from "react";
import {
    AppBar, createStyles,
    IconButton, Theme,
    Toolbar,
    withStyles,
    WithStyles
} from "@material-ui/core";
import SortIcon from "@material-ui/icons/Sort";
import ExpandMoreIcon from "@material-ui/icons/ExpandMore";

function Header({ classes }: HeaderProps) {
    const [checked, setChecked] = useState(false);
    useEffect(() => {
        setChecked(true);
    }, []);
    return (
        <div className={classes.root} id="header">
            <AppBar className={classes.appbar} elevation={0}>
                <Toolbar className={classes.appbarWrapper}>
                    <h1 className={classes.appbarTitle}>
                        <span className={classes.colorText}>Bookeroo.</span>
                    </h1>
                    <IconButton>
                        <SortIcon className={classes.icon}/>
                    </IconButton>
                </Toolbar>
            </AppBar>
            <div className={classes.container}>
                <h1 className={classes.title}>
                    Welcome to <br/>
                    <span className={classes.colorText}>Bookeroo.</span>
                </h1>
                <IconButton>
                    <ExpandMoreIcon className={classes.goDown}/>
                </IconButton>
            </div>
        </div>
    )
}

const styles = (theme: Theme) =>
    createStyles({
        root: {
            display: "flex",
            justifyContent: "center",
            alignItems: "center",
            height: "100vh",
            fontFamily: "Nunito",
        },
        appbar: {
            background: "none",
        },
        appbarWrapper: {
            width: "80%",
            margin: "0 auto",
        },
        appbarTitle: {
            flexGrow: 1,
        },
        icon: {
            color: "#F78888",
            fontSize: "2rem",
        },
        colorText: {
            color: "#5DA2D5",
        },
        container: {
            textAlign: "center",
        },
        title: {
            color: "#F8E9A1",
            fontSize: "4.5rem",
        },
        goDown: {
            color: "#A8D0E6",
            fontSize: "4rem",
        },
    });


type HeaderProps = WithStyles<typeof styles>;

export default withStyles(styles)(Header);

