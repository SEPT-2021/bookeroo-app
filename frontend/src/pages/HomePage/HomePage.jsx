import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { CssBaseline } from '@material-ui/core';
import bgi from "../../assets/backgroundBook.jpg";

const useStyles = makeStyles((theme) => ({
    root: {
        minHeight: '100vh',
        backgroundImage: `url(${process.env.PUBLIC_URL + 'bgi'})`,
        backgroundRepeat: 'no-repeat',
        backgroundSize: 'cover',
    },
}));
export default function App() {
    const classes = useStyles();
    return (
        <div className={classes.root}>
            <CssBaseline />
        </div>
    );
}