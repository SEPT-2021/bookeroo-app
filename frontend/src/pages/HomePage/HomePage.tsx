import React from "react";
import {
  createStyles,
  CssBaseline,
  withStyles,
  WithStyles,
} from "@material-ui/core";
import Header from "../../components/Header/Header";
import backgroundImage from "../../assets/backgroundBook.png";

function HomePage({ classes }: HomePageProps) {
  return (
    <div className={classes.root}>
      <CssBaseline />
      <Header />
    </div>
  );
}

const styles = () =>
  createStyles({
    root: {
      minHeight: "100vh",
      backgroundImage: `url(${process.env.PUBLIC_URL + backgroundImage})`,
      backgroundRepeat: "no-repeat",
      backgroundSize: "cover",
    },
  });

type HomePageProps = WithStyles<typeof styles>;

export default withStyles(styles)(HomePage);
