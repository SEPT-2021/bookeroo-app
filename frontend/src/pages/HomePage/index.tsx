import React from "react";
import { createStyles, withStyles, WithStyles } from "@material-ui/core";
import Header from "../../components/Header";
import backgroundImage from "../../assets/books/backgroundBook2.png";
import Footer from "../../components/Footer";

function HomePage({ classes }: HomePageProps) {
  return (
    <div className={classes.root}>
      <Header />
      <Footer />
    </div>
  );
}

const styles = () =>
  createStyles({
    root: {
      minHeight: "100vh",
      backgroundImage: `url(${process.env.PUBLIC_URL + backgroundImage})`,
      backgroundRepeat: "no-repeat",
      backgroundSize: "cover",
    },
  });

type HomePageProps = WithStyles<typeof styles>;

export default withStyles(styles)(HomePage);
