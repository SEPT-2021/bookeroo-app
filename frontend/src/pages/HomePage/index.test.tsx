import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup } from "@testing-library/react";
import { Button } from "@material-ui/core";
import { Router } from "@material-ui/icons";
import HomePage from "./index";
import Header from "../../components/Header";
import Footer from "../../components/Footer";

afterEach(cleanup);
describe("Header matches snapshot", () => {
  test("render Header", () => {
    const { container } = render(
      <Router>
        <Header />
      </Router>
    );
    expect(container).toMatchSnapshot();
  });
});

describe("Render component Footer", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Footer />, div);
    expect(div).toMatchSnapshot();
  });
});

describe("Button render matches snapshot", () => {
  test("renders button", () => {
    const button = render(<Button />);
    expect(button).toMatchSnapshot();
  });
});

describe("HomePage matches snapshot", () => {
  test("render HomePage", () => {
    const { container } = render(
      <Router>
        <HomePage />
      </Router>
    );
    expect(container).toMatchSnapshot();
  });
});
