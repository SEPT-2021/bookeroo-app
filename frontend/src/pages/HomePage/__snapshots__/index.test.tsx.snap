// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render matches snapshot renders button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      />
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Header matches snapshot render Header 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M20.2 5.9l.8-.8C19.6 3.7 17.8 3 16 3s-3.6.7-5 2.1l.8.8C13 4.8 14.5 4.2 16 4.2s3 .6 4.2 1.7zm-.9.8c-.9-.9-2.1-1.4-3.3-1.4s-2.4.5-3.3 1.4l.8.8c.7-.7 1.6-1 2.5-1 .9 0 1.8.3 2.5 1l.8-.8zM19 13h-2V9h-2v4H5c-1.1 0-2 .9-2 2v4c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2v-4c0-1.1-.9-2-2-2zM8 18H6v-2h2v2zm3.5 0h-2v-2h2v2zm3.5 0h-2v-2h2v2z"
    />
  </svg>
</div>
`;

exports[`HomePage matches snapshot render HomePage 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M20.2 5.9l.8-.8C19.6 3.7 17.8 3 16 3s-3.6.7-5 2.1l.8.8C13 4.8 14.5 4.2 16 4.2s3 .6 4.2 1.7zm-.9.8c-.9-.9-2.1-1.4-3.3-1.4s-2.4.5-3.3 1.4l.8.8c.7-.7 1.6-1 2.5-1 .9 0 1.8.3 2.5 1l.8-.8zM19 13h-2V9h-2v4H5c-1.1 0-2 .9-2 2v4c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2v-4c0-1.1-.9-2-2-2zM8 18H6v-2h2v2zm3.5 0h-2v-2h2v2zm3.5 0h-2v-2h2v2z"
    />
  </svg>
</div>
`;

exports[`Render component Footer renders without crashing 1`] = `
<div>
  <div
    class="Footer-footer-1"
  >
    <footer>
      <p>
         2021 Â© SEPT Bookeroo. All rights reserved.
      </p>
    </footer>
  </div>
</div>
`;
